src/coredump/coredump.c

```diff
@@ -1119,9 +1214,23 @@ static int process_kernel(int argc, char* argv[]) {
 
         if (get_process_root(pid, &t) >= 0) {
                 core_root = strjoina("COREDUMP_ROOT=", t);
-                free(t);
 
                 IOVEC_SET_STRING(iovec[n_iovec++], core_root);
+
+                /* If the process' root is "/", then there is a chance it has
+                 * mounted own root and hence being containerized. */
+                if (strcmp(t, "/") == 0) {
```

Comment:
This duplicated `free(t)` is somewhat ugly. Maybe something like this instead:
```
bool root;

root = path_equal(t, "/");
free(t);
...
if (root) ...
```

`path_equal` is preferred to standard string comparisons for paths.